cpp/dolfin/mesh/DistributedMeshTools.cpp

```diff
@@ -1101,13 +1102,17 @@ void DistributedMeshTools::init_facet_cell_connections(Mesh& mesh)
 EigenRowArrayXXd
 DistributedMeshTools::reorder_points_by_global_indices(const Mesh& mesh)
 {
-  return reorder_values_by_global_indices(mesh.mpi_comm(),
+  return reorder_values_by_global_indices<double>(mesh.mpi_comm(),
                                           mesh.geometry().points(),
                                           mesh.geometry().global_indices());
 }
 //-----------------------------------------------------------------------------
-EigenRowArrayXXd DistributedMeshTools::reorder_values_by_global_indices(
-    MPI_Comm mpi_comm, const Eigen::Ref<const EigenRowArrayXXd>& values,
+template <typename Scalar>
+Eigen::Array<Scalar, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>
```

Comment:
`Scalar` is confusing to me. My immediate reaction is that it's a concrete type, as `PetscScalar` is. 

I would suggest using `T` or `X`.